### PR TITLE
doc/reference: move the settings doc higher

### PR DESCRIPTION
**Pull request by nvlsianpu**
Settings documentation was placed under storage paragraph
in documentation tree, which suggested that it was another
storage solution. In fact settings uses storage modules as
its persistent storage back-end.

This patch creates dedicated run-time configuration paragraph
which aim to explain existence (and emphasize) of common run-time
configuration system in Zephyr-RTOS and rationale for using it.
The Settings paragraph is placed under this paragraph as it is only
sub-module which implements the solution.

Rationale for this movement i doc is that user often can't find this dedicated solution at the first glance, so they either ask on slack or get now on their PR comments.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: doc/reference/runtime_conf/index.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. _runtime_config_reference:
+
+
+Runtime Configuration System
+############################
+
+Zephyr-RTOS provide common submodule called 'settings' for store persistently
```
Comment (dbkinder) [inline doc/reference/runtime_conf/index.rst]: Fixing some misspellings and grammar:
```
Zephyr RTOS provides a submodule called 'settings' used
persistent storage such as saving and loading runtime
configuration data. A variety of storage implementations
are provided behind a common API using FCB, NVS, or a
file system.  These different implementations give the
application developer flexibility to select an appropriate
storage medium, and even change it later as needs change.

The runtime configuration system is used by various Zephyr
components and can be used simultaneously by user applications.
```

Did I get the details right?

**Review by dbkinder — CHANGES_REQUESTED** [review]
Making the API more obvious is a good ides.  I've got a suggested rewrite of the description.

File: doc/reference/runtime_conf/index.rst (hunk 1 above)
Comment (nvlsianpu) [reply doc/reference/runtime_conf/index.rst]: ^^ looks fine to me

Comment (dbkinder) [reply doc/reference/runtime_conf/index.rst]: OK, if you update your PR with this change, the doc is ready...

Comment (nvlsianpu) [reply doc/reference/runtime_conf/index.rst]: @dbkinder update

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Comment by nashif** [comment]
@nvlsianpu  please rebase

**Comment by nvlsianpu** [comment]
@nashif @carlescufi rebased

**Review by ioannisg — APPROVED** [review]

**Pull request merged**